poetry/console/commands/shell.py

```diff
@@ -0,0 +1,51 @@
+from os import environ
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        # This variable will be set if the current shell was spawned by poetry
+        current_venv = environ.get("POETRY_ACTIVE")
+
+        # Alternatively, a virtual environment might have already been
+        # activated in some other way
+        if not current_venv and not self.venv.is_venv():
```

Comment:
I don't get it... Shouldn't you check for `... and self.venv.is_venv()` (without not)? If `venv.is_venv` is `False` then there is no guarantee that you will find `environ.get("VIRTUAL_ENV")`... or am I missing something?

Reply:
Yeah, it's a bit confusing, but that's what I was ranting about a couple of days ago. If you manually activate a virtualenv and then run some poetry command that inherits from `VenvCommand`, its `venv` attribute will be set to an empty `Venv` instance, with `venv=None`. I'm still not sure that is correct.

Reply:
Ook.. I get it. So the artifacts of the poetry being run in an activated venv is that there is `VIRTUAL_ENV` env variable set and the built Venv instance has an attribute `venv` which is `None`. Semantically it's rather counter intuitive... Couldn't we better this design on this occasion? @sdispater what are your thoughts? I'll look into it and try to compe up with some proposition - it would be cool if `Venv` represented the target virtual environment even if it was activsted from somehere else

Reply:
I must admit that the `is_venv()` method name is a bit unfortunate and I understand that it might be misleading. But like you already figured out, it's here to check if we are already inside a virtualenv (or no virtualenv at all) or if we are using the poetry-managed virtualenv.

> it would be cool if Venv represented the target virtual environment even if it was activsted from somehere else

I agree. And in this case the `is_venv()` would only return true if we are not using virtualenvs at all (which should not happen very often since it suppose the use of the `settings.virtualenvs.create` and setting it to `false`).

However, I think it should be done in a separate PR and not make this one too complex.  